Testscripts/Linux/XDPForwardingTest.sh

```diff
@@ -0,0 +1,170 @@
+# in forwarding configuration on forwarder VM and checks how many packets received at the receiver interface
+# by running xdpdump application in drop configuration (number of packets received == number of packets dropped).
+
+
+packetCount=10000000
+nicName='eth1'
+packetFwdThreshold=90
+
+function convert_MAC_to_HEXArray(){
+        while IFS=':' read -ra ADDR; do
+                size=$((${#ADDR[@]} - 1))
+                MACarr=$(printf '0x%s\n' ${ADDR[$i]})
+                for i in $(seq 1 $size);
+                do
+                        MACarr="$MACarr, $(printf '0x%s\n' ${ADDR[$i]})";
+                done
+        done <<< "$1"
+        echo "$MACarr"
+}
+
+function configure_XDPDUMP_TX(){
+        LogMsg "Configuring TX Setup"
+        get_ip_command="/sbin/ifconfig $nicName | grep 'inet' | cut -d: -f2"
+        get_mac_command="/sbin/ifconfig $nicName | grep -o -E '([[:xdigit:]]{1,2}:){5}[[:xdigit:]]{1,2}'"
+        forwarderIP=$((ssh $forwarder $get_ip_command) | awk '{print $2}')
+        LogMsg "Forwarder IP: $forwarderIP"
+        receiverIP=$((ssh $receiver $get_ip_command) | awk '{print $2}')
+        LogMsg "Receiver IP: $receiverIP"
+        forwarderMAC=$(ssh $forwarder $get_mac_command)
+        LogMsg "Forwarder MAC: $forwarderMAC"
+        receiverMAC=$(ssh $receiver $get_mac_command)
+        LogMsg "Receiver MAC: $receiverMAC"
+
+        #formatting MAC and IP address as needed in xdpdump file.
+        forwarderIP1=$(echo $forwarderIP | sed "s/\./\, /g")
+        receiverIP1=$(echo $receiverIP | sed "s/\./\, /g")
+        forwarderMAC1=$(convert_MAC_to_HEXArray $forwarderMAC)
+        receiverMAC1=$(convert_MAC_to_HEXArray $receiverMAC)
+        xdpdumpFileName=bpf-samples/xdpdump/xdpdump_kern.c
+
+        LogMsg "Updating $xdpdumpFileName file with forwarding setup on $forwarder"
+        commandMACS="sed -i 's/unsigned char newethsrc \[\] = { 0x00, 0x22, 0x48, 0x4c, 0xc4, 0x4d };/unsigned char newethsrc \[\] = { ${forwarderMAC1} };/g' ${xdpdumpFileName}"
+        ssh $forwarder $commandMACS
+        commandMACD="sed -i 's/unsigned char newethdest \[\] = { 0x00, 0x22, 0x48, 0x4c, 0xc0, 0xfd };/unsigned char newethdest \[\] = { ${receiverMAC1} };/g' ${xdpdumpFileName}"
+        ssh $forwarder $commandMACD
+        LogMsg "Updated Source &  Destination MAC address in $xdpdumpFileName on $forwarder"
+        commandIPS="sed -i 's/__u8 newsrc \[\] = { 10, 0, 1, 5 };/__u8 newsrc \[\] = { ${forwarderIP1} };/g' ${xdpdumpFileName}"
+        ssh $forwarder $commandIPS
+        commandIPD="sed -i 's/__u8 newdest \[\] = { 10, 0, 1, 4 };/__u8 newdest \[\] = { ${receiverIP1} };/g' ${xdpdumpFileName}"
+        ssh $forwarder $commandIPD
+        LogMsg "Updated Source &  Destination IP address in $xdpdumpFileName on $forwarder"
+}
+
+UTIL_FILE="./utils.sh"
+
+# Source utils.sh
+. ${UTIL_FILE} || {
+    echo "ERROR: unable to source ${UTIL_FILE}!"
+    echo "TestAborted" > state.txt
+    exit 0
```

Comment:
Please check the indentation, not the same with others.

Reply:
Indentation is 4 spaces